lib/upipe-dvbcsa/upipe_dvbcsa_bs_decrypt.c

```diff
@@ -0,0 +1,527 @@
+/*
+ * Copyright (C) 2017 OpenHeadend S.A.R.L.
+ *
+ * Authors: Arnaud de Turckheim
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject
+ * to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <upipe-dvbcsa/upipe_dvbcsa_bs_decrypt.h>
+#include <upipe-dvbcsa/upipe_dvbcsa_common.h>
+
+#include <upipe/upipe.h>
+#include <upipe/upump.h>
+#include <upipe/uclock.h>
+#include <upipe/uref_block.h>
+#include <upipe/uref_clock.h>
+
+#include <upipe/upipe_helper_upipe.h>
+#include <upipe/upipe_helper_urefcount.h>
+#include <upipe/upipe_helper_void.h>
+#include <upipe/upipe_helper_output.h>
+#include <upipe/upipe_helper_uclock.h>
+#include <upipe/upipe_helper_upump_mgr.h>
+#include <upipe/upipe_helper_upump.h>
+#include <upipe/upipe_helper_input.h>
+
+#include <bitstream/mpeg/ts.h>
+#include <dvbcsa/dvbcsa.h>
+
+#include "common.h"
+
+/** expected input flow format */
+#define EXPECTED_FLOW_DEF "block.mpegts."
+/** dvbcsa decrypt latency */
+#define DVBCSA_LATENCY  (UCLOCK_FREQ / 100)
```

Comment:
Add a comment to say this value comes from a statistic on normal hardware.

Reply:
Fixed